### PR TITLE
Android: Rework settings UI

### PR DESCRIPTION
**Pull request by ghost**
This adds icons to the settings page and tweaks the padding 
Feel free to change the description labels of the settings   

- Credit to @ishan09811

### CODE REVIEW

**Comment by BlurrySquire** [comment]
Can't test at the moment, but with the option for resetting the config is there a confirmation pop-up? I think many users would appreciate it being added if there isn't an option.

**Comment by BlurrySquire** [comment]
> Can't test at the moment, but with the option for resetting the config is there a confirmation pop-up? I think many users would appreciate it being added if there isn't an option.

Just caught up in the discord and realised that I got a response when asking if it had a pop-up. It does.

**Comment by OpenSauce04** [comment]
Please provide a side-by-side comparison between the current release and your proposed changes

**Comment by ghost** [comment]
> Please provide a side-by-side comparison between the current release and your proposed changes

## Current Design 




## Proposed Design

**Comment by OpenSauce04** [comment]
The descriptions below each setting are redundant and add unnecessary visual noise.

They should be removed.

File: src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SubmenuViewHolder.kt [hunk 1]
```diff
@@ -16,13 +17,27 @@ class SubmenuViewHolder(val binding: ListItemSettingBinding, adapter: SettingsAd
...
+                ResourcesCompat.getDrawable(
+                    binding.icon.resources,
+                    item.iconId,
+                    binding.icon.context.theme
+                )
+            )
+        } else {
+            binding.icon.visibility = View.GONE
+        }
+
```
Comment (OpenSauce04) [inline src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SubmenuViewHolder.kt]: Invert this if statement.

File: src/android/app/src/main/java/io/github/lime3ds/android/utils/ViewUtils.kt [hunk 2]
```diff
@@ -33,4 +34,28 @@ object ViewUtils {
...
+            }
+            if (top != -1) {
+                topMargin = top
+            }
+            if (right != -1) {
+                rightMargin = right
+            }
+            if (bottom != -1) {
+                bottomMargin = bottom
+            }
```
Comment (OpenSauce04) [inline src/android/app/src/main/java/io/github/lime3ds/android/utils/ViewUtils.kt]: This code should lose the braces.

**Review by OpenSauce04 — CHANGES_REQUESTED** [review]

**Comment by OpenSauce04** [comment]
I will re-review this in the morning.

File: src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/RunnableViewHolder.kt [hunk 3]
```diff
@@ -18,6 +19,19 @@ class RunnableViewHolder(val binding: ListItemSettingBinding, adapter: SettingsA
...
+            binding.icon.setImageDrawable(
+                ResourcesCompat.getDrawable(
+                    binding.icon.resources,
+                    item.iconId,
+                    binding.icon.context.theme
+                )
+            )
+        } else {
+            binding.icon.visibility = View.GONE
+        }
```
Comment (OpenSauce04) [inline src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/RunnableViewHolder.kt]:
```suggestion
        if (item.iconId == 0) {
            binding.icon.visibility = View.GONE
        } else {
            binding.icon.visibility = View.VISIBLE
            binding.icon.setImageDrawable(
                ResourcesCompat.getDrawable(
                    binding.icon.resources,
                    item.iconId,
                    binding.icon.context.theme
                )
            )
        }
```

File: src/android/app/src/main/java/io/github/lime3ds/android/utils/ViewUtils.kt [hunk 4]
```diff
@@ -33,4 +34,20 @@ object ViewUtils {
...
+        top: Int = -1,
+        right: Int = -1,
+        bottom: Int = -1
+    ) {
+        val layoutParams = this.layoutParams as ViewGroup.MarginLayoutParams
+        layoutParams.apply {
+            if (left != -1) leftMargin = left
+            if (top != -1) topMargin = top
+            if (right != -1) rightMargin = right
+            if (bottom != -1) bottomMargin = bottom
```
Comment (OpenSauce04) [inline src/android/app/src/main/java/io/github/lime3ds/android/utils/ViewUtils.kt]:
```suggestion
            if (left != -1)
                leftMargin = left
            if (top != -1)
                topMargin = top
            if (right != -1)
                rightMargin = right
            if (bottom != -1)
                bottomMargin = bottom
```

File: src/android/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -294,6 +294,7 @@
     <string name="learn_more">Learn More</string>
     <string name="close">Close</string>
     <string name="reset_to_default">Reset to Default</string>
+    <string name="reset_to_default_description">Resets all settings</string>
```
Comment (OpenSauce04) [inline src/android/app/src/main/res/values/strings.xml]: Seems this string was left in by accident when removing the descriptions.

**Review by OpenSauce04 — CHANGES_REQUESTED** [review]

**Comment by OpenSauce04** [comment]
All LGTM.

@kleidis Any issues with the changes in 60a00a9? If not, this is ready to merge.
As far as I can tell there is no user-facing change here, it simply cleans up unnecessary code:

**Review by OpenSauce04 — APPROVED** [review]

**Comment by ghost** [comment]
> @kleidis Any issues with the changes in 60a00a9? If not, this is ready to merge.

Seems fine , go ahead

**Comment by Polarzincomfrio** [comment]
Late but decided to give it a check anyways for redundancy, works fine on A54 and looks better than before, nice work

**Review by ayrudev — COMMENTED** [review]
Late but decided to review anyway, it works fine on my A54 and looks nice, nothing to note code-wise

**Pull request merged**